src/fruity/fruity-host-session.vala

```diff
@@ -497,7 +497,8 @@ namespace Frida {
 			var device_info = yield Fruity.DeviceInfoService.open (channel_provider, cancellable);
 			var processes = yield device_info.enumerate_processes (cancellable);
 
-			var process = processes.first_match (p => p.foreground_running && p.real_app_name != SPRINGBOARD_PATH);
+			var process = processes.first_match (p => p.foreground_running && p.real_app_name != SPRINGBOARD_PATH &&
+					!p.real_app_name.contains (".appex/"));

```

Comment:
Is there some metadata we could use instead of sniffing the path? IIRC we currently don't expose all of the properties.

Reply:
interesting! checking

Reply:
ok, i double checked and there's a promising `isApplication` field in the process object, but unfortunately that's `1` also for some of the appex (like Today widgets)

Reply:
maybe it's worth exposing and checking it anyway to shrink the number of string comparisons to what's strictly needed

Reply:
Ah that would be great actually!  Wanna amend this one or leave it for later?

Reply:
i'll just add here if you're fine with that

Reply:
Yep sounds great!